pgjdbc/src/test/java/org/postgresql/test/jdbc2/BatchExecuteTest.java

```diff
@@ -1253,6 +1260,38 @@ public void testBatchWithTwoMultiInsertStatements() throws SQLException {
     }
   }
 
+  @Test
+  public void testBatchWithGeneratedKeys() throws SQLException {
+    PreparedStatement pstmt = null;
+
+
+    try {
+      pstmt = con.prepareStatement("INSERT INTO testbatchserial (col1) VALUES (?)", new String[] {"id"});
+
+      for ( int i = 0; i < 10; i++ ){
+        pstmt.setInt(1,i);
+        pstmt.execute();
+        ResultSet genkeys = pstmt.getGeneratedKeys();
+        Assert.assertTrue(genkeys.next());
+        Assert.assertEquals(i+1, genkeys.getLong(1));
+      }
+
```

Comment:
This exact (blank) line is "wrong" =)

The statement should be closed in order to get reused. So please close the `pstmt` in between, and it should do the trick.